src/setfit/onnx.py

```diff
@@ -0,0 +1,212 @@
+        raise ImportError(msg)
+
+    # Check to see that the head has a coef_
+    if not hasattr(model_head, "coef_"):
+        raise ValueError(
+            "Head must have coef_ attribute check that this is supported by your model and the model has been fit."
+        )
+
+    # Determine the initial type and the shape of the output.
+    input_shape = (None, *model_head.coef_.shape[1:])
+    dtype = guess_data_type(model_head.coef_, shape=input_shape)[0][1]
+    dtype.shape = input_shape
+
+    # If the datatype of the model is double we need to cast the outputs
+    # from the setfit model to doubles for compatibility inside of ONNX.
+    if type(dtype) == onnxconverter_common.data_types.DoubleTensorType:
+        # TODO:: TALK ABOUT FLOAT CONVERSION ISSUES.
+        sk_model = Pipeline([("castdouble", CastTransformer(dtype=np.double)), ("head", model_head)])
+    else:
+        sk_model = model_head
+
+    # Convert sklearn head into ONNX format
+    initial_type = [("model_head", dtype)]
+    onx = convert_sklearn(
+        sk_model, initial_types=initial_type, target_opset=opset, options={id(sk_model): {"zipmap": False}},
+    )
+
+    return onx
+
+
+def export_onnx(
+    model_body_sentence_transformer, model_head, opset: int, output: str, ignore_ir_version: bool = True,
+):
+    """
+    Export a PyTorch backed setfit model to ONNX Intermediate Representation.
+
+    Args:
+        model_body_sentence_transformer: The model_body from a SetFitModel.model_body. This should be a
+            SentenceTransformer.
+        model_head: The model_head from SetFitModel.model_head. This can be either a dense
+            layer SetFitHead or a Sklearn estimator.
+        opset: The actual version of the ONNX operator set to use.  The final opset used
+            might be lower. ONNX will use the highest version supported by both the
+            sklearn head and the model body. If versions can't be rectified an error
+            will be thrown.
+        output: The path where will be stored the generated ONNX model. If the head is an
+            sklearn.estimator the models will be saved in
+        ignore_ir_version: Whether to ignore the IR version used in sklearn. The version is often missmatched
+            with the transformer models. Setting this to true coerces the versions to be the same. This might
+            cause errors but in practice works.  If this is set to False you need to ensure that the IR versions
+            align between the transformer and the sklearn onnx representation.
+    Returns:
+    """
+
+    # Load the model and get all of the parts.
+    transformer = model_body_sentence_transformer._modules["0"]
+    tokenizer = transformer.tokenizer
+    max_length = transformer.max_seq_length
+    model_body = transformer.auto_model
+    model_pooler = model_body_sentence_transformer._modules["1"]
+    model_head = model.model_head
```

Comment:
Just highlighting this line; with the new commits here, there is no `model` variable anymore.